operationsgateway_api/src/experiments/experiment.py

```diff
@@ -72,16 +72,40 @@ async def store_experiments(self) -> None:
         that haven't yet been inserted into the database
         """
 
+        inserted_ids = []
         for experiment in self.experiments:
-            await MongoDBInterface.update_one(
+            experiment_data = experiment.dict(
+                by_alias=True,
+                exclude_unset=True,
+                exclude={"id_"},
+            )
+
+            update_result = await MongoDBInterface.update_one(
                 "experiments",
-                {"_id": experiment.id_},
-                {"$set": experiment.dict(by_alias=True)},
+                experiment_data,
+                {"$set": experiment_data},
                 upsert=True,
             )
+            # This means the document was updated, not inserted. The _id needs to be
+            # found to be used as a response
+            if update_result.upserted_id is None:
+                updated_experiment = await MongoDBInterface.find_one(
+                    "experiments",
+                    experiment_data,
+                )
+
+                log.debug(
+                    "Experiment was updated, found experiment _id: %s",
+                    str(updated_experiment["_id"]),
+                )
+                inserted_ids.append(f"Updated {str(updated_experiment['_id'])}")
+            else:
+                inserted_ids.append(str(update_result.upserted_id))
 
         await self._update_modification_time()
 
+        return inserted_ids
```

Comment:
The return type of the function needs updating now that something is being returned

Reply:
Fixed in 592a500182053d2aff91fdaabb6d09d92ddc6af3